Stratis.Bitcoin/Features/BlockStore/LoopSteps/BlockStoreInnerStepContext.cs

```diff
@@ -16,18 +18,35 @@ namespace Stratis.Bitcoin.Features.BlockStore.LoopSteps
     /// </summary>
     public sealed class BlockStoreInnerStepContext
     {
+        /// <summary>Number of milliseconds to wait after each failed attempt to get a block from the block puller.</summary>
+        internal const int StallDelayMs = 100;
+
+        /// <summary><see cref="DownloadStack"/> is flushed to the disk if more than this amount of milliseconds passed since the last flush was made.</summary>
+        internal const int MaxDownloadStackFlushTimeMs = 20 * 1000; 
+
         /// <summary>Instance logger.</summary>
         private readonly ILogger logger;
 
         /// <summary>Factory for creating loggers.</summary>
         private readonly ILoggerFactory loggerFactory;
 
-        public BlockStoreInnerStepContext(CancellationToken cancellationToken, BlockStoreLoop blockStoreLoop, ILoggerFactory loggerFactory)
+        /// <summary>Provider of time functions.</summary>
+        internal readonly IDateTimeProvider DateTimeProvider;
```

Comment:
Make this a property with a private setter:

`internal readonly IDateTimeProvider DateTimeProvider { get; private set;}`